syntax = "proto3";

package user;

// User service definition
service UserService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
  rpc GetMultipleUsers(GetMultipleUsersRequest) returns (GetMultipleUsersResponse);
}

// Messages for GetUser
message GetUserRequest {
  int32 user_id = 1;
}

message GetUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

// Messages for CreateUser
message CreateUserRequest {
  string name = 1;
  string email = 2;
  string phone = 3;
}

message CreateUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

// Messages for GetMultipleUsers
message GetMultipleUsersRequest {
  repeated int32 user_ids = 1;
}

message GetMultipleUsersResponse {
  bool success = 1;
  string message = 2;
  repeated User users = 3;
}

// User model
message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string phone = 4;
  int64 created_at = 5;
}